### ISSUE TITLE
Deprecate `Linear` builder?

### ISSUE THREAD

**Issue by ablaom** [FluxML/MLJFlux.jl#253]
The `MLP` builder added more recently renders it redundant.

**Comment by EssamWisam** [comment]
Agree. Think should also deprecate `Short` because it's also a special case of `MLP` and forces specifically three hidden layers (also quite hard to infer the functionality from the name; a 3 layer network can theoretically even approximate discontinuous functions to arbitrary accuracy).

**Comment by ablaom** [comment]
'Short' includes a dropout  layer and for this reason is needed in tests, if I remember correctly. I agree the name is not great and we could deprecate to "testing only".

**Comment by EssamWisam** [comment]
Sheds light on the possibility of extending MLPs to handle dropout (and maybe batch norm because it's just as common). For instance, for drop out, user can pass a list of probabilities that correspond to the hidden layers and 0 would mean no dropout.
